roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,181 @@
+        all available storage from its VG, instead expanding as needed; so in considering
+        available space, it is necessary to include space remaining in the VG (which is
+        unfortunately not reported by the storage driver in docker info).
+        Usage of the LV is divided into space allocated to data and metadata (which also
+        expand automatically), either of which could run out of space while the other is
+        fine; so both must be checked.
+        """
+        vals = dict(
+            vg_free=self._get_vg_free(driver_status.get("Pool Name"), task_vars),
+            data_used=driver_status.get("Data Space Used"),
+            data_total=driver_status.get("Data Space Total"),
+            metadata_used=driver_status.get("Metadata Space Used"),
+            metadata_total=driver_status.get("Metadata Space Total"),
+        )
+
+        for key, value in vals.copy().items():
+            try:
+                vals[key + "_bytes"] = self._convert_to_bytes(value)
+            except ValueError as err:  # unlikely to see this from our inputs
+                return {
+                    "failed": True,
+                    "values": vals,
+                    "msg": "Could not interpret {} value '{}' as bytes: {}".format(key, value, str(err))
+                }
+
+        try:
+            vals["data_threshold"] = float(get_var(task_vars, "max_thinpool_data_usage_percent",
+                                                   default=self.max_thinpool_data_usage_percent))
+            vals["metadata_threshold"] = float(get_var(task_vars, "max_thinpool_metadata_usage_percent",
+                                                       default=self.max_thinpool_meta_usage_percent))
+        except ValueError as err:
+            return {
+                "failed": True,
+                "msg": "Unable to convert thinpool data usage limit to float: {}".format(str(err))
+            }
+
+        vals["data_pct_used"] = 100 * vals["data_used_bytes"] / (
+            vals["data_total_bytes"] + vals["vg_free_bytes"])
+        vals["metadata_pct_used"] = 100 * vals["metadata_used_bytes"] / (
+            vals["metadata_total_bytes"] + vals["vg_free_bytes"])
+        msg = ""
+        if vals["data_threshold"] < vals["data_pct_used"]:
+            vals["failed"] = True
+            msg = (
+                "Docker thinpool data usage percentage {:.1f} "
+                "is higher than threshold {:.1f}."
+            ).format(vals["data_pct_used"], vals["data_threshold"])
+        if vals["metadata_threshold"] < vals["metadata_pct_used"]:
+            vals["failed"] = True
+            msg = msg + "\n" if msg else ""
+            msg += (
+                "\nDocker thinpool metadata usage percentage {:.1f} "
+                "is higher than threshold {:.1f}."
+            ).format(vals["metadata_pct_used"], vals["metadata_threshold"])
+
+        vals["msg"] = msg or "Thinpool usage is below thresholds."
```

Comment:
Aha, didn't realize that... I'm used to see a `'msg'` key accompanying a `'failed': True`, didn't think of the other case ;-)